### ISSUE TITLE
multiprocessing pool causes confusing behavior when attempting simplest possible case

### ISSUE THREAD

**Issue by untzag** [wright-group/NISE#2]
need to be under if main: to run scans in default operation

if this is how we want it to be I guess that is fine, but we should if nothing else change all the examples in lib so they aren't broken

**Comment by ddkohler** [comment]
Agreed that this should not be how it works.  I'm pretty sure there is a workaround, but it will take further searching.  Let me know if you find something.

wrt lib examples, I think those examples should be replaced with iPython notebooks.  The nice thing about using iPython is that is seems like you can forego the name-main clause and it still works.  It's secret is probably key to fixing this.

Related:  where do we store the iPython notebooks for said examples?

**Comment by untzag** [comment]
I propose that examples in lib/testing_scripts become exclusively the domain of those working inside of NISE. 

You could create a new top-level folder 'examples' that contains those ipython notebooks.

Regarding how to defeat the main: issue, I find [this answer](http://stackoverflow.com/a/20222706) enlightening. I think the trick of ipython notebooks is that each cell is run as if pasted into the ipython console. They don't run like a normal module on input.

I think we'll never be able to make a script you can simply run from spyder and have work without main:. However, we could structure our script like this:

``` python

def run():
    import NISE 
    trive.set_coord()
    out1 = trive.exp.scan()

```

The user would set up their run settings in spyder to execute externally and interact after execution. The console would open and do nothing, but the scan would execute without errors once the user typed run().

I'm _really_ liking this issues system for keeping organized discussions on these issues.
